OpenRA.Game/Map/MapCache.cs

```diff
@@ -174,7 +174,8 @@ void LoadAsyncInternal()
 					if (bitmap == null)
 					{
 						createdPreview = true;
-						bitmap = Minimap.RenderMapPreview(modData.DefaultRules.TileSets[p.Map.Tileset], p.Map, modData.DefaultRules, true);
+						var map = new Map(p.Path);
```

Comment:
This is super not thread-safe. There's way too much shared state being shared here.